### DISCUSSION TITLE
Completely reset webpack config when creating a child compiler

### DISCUSSION THREAD

**Discussion by LukasBombach** [webpack/webpack#16670] [Q&A] [answered]
We need inline-JavaScript to initialize advertisements, tracking and some other things. We're also using Next.js, which does not really offer good tools to create inline JavaScript (other than providing code as plain-text).

So we built our own tool, that uses a child compiler to render some parts of our code and inline its results during the SSR process.

This method has a flaw: Since we are doing this on the server, the child compiler uses a webpack config that targets node, rather than the browser. We already overwrote some settings, but not all. This setup is a bit fragile, we have unknown unknowns in regards to which settings we need to overwrite.

It would be nice to be able to use a child compiler but start from a blank config and be able to provide a clean browser-targeted config. Is there any way to do this?

**Comment by alexander-akait** [comment, marked as answer]
Sorry, there is no way, because a child compilation inherts a parent configuration, you need to reset all options manually

**Reply by LukasBombach** [reply]
I see thank you very much!
